app/lib/dartdoc/dartdoc_page.dart

```diff
@@ -0,0 +1,419 @@
+
+  /// Link to the breadcrumb, `null` if this the current page
+  final String? href;
+
+  Breadcrumb({required this.title, required this.href});
+}
+
+/// A dartdoc page parsed as sanitized html.
+///
+///
+/// This assumes a page layout along the lines of:
+/// ```
+///           [header]
+/// ------------------------------
+///        |           |
+/// [left] | [content] | [right]
+///        |           |
+/// ------------------------------
+///           [footer]
+/// ```
+class DartDocPage {
+  final String title;
+  final String description;
+  final List<Breadcrumb> breadcrumbs;
+  final String left;
+  final String right;
+  final String content;
+
+  DartDocPage({
+    required this.title,
+    required this.description,
+    required this.breadcrumbs,
+    required this.left,
+    required this.right,
+    required this.content,
+  });
+
+  static DartDocPage parse(String rawHtml) {
+    final document = html_parser.parse(rawHtml);
+
+    // TODO: Create a variant of sanitizeHtml that consumes nodes from
+    //       package:html, so that we don't have re-parse everything :/
+    String sanitize(String? html) => sanitizeHtml(
+          html ?? '',
+          allowClassName: (_) => true,
+          allowElementId: (_) => true,
+          addLinkRel: (href) {
+            final u = Uri.tryParse(href);
+            if (u != null && !u.shouldIndicateUgc) {
+              // TODO: Determine if there is a better way to do this.
+              //       It's probably reasonable that internal links don't
+              //       required ugc + nofollow.
+              return [];
+            }
+            // Add ugc and nofollow if the host isn't one of ours.
+            return ['ugc', 'nofollow'];
+          },
+        );
+
+    // HACK: Adding markdown-body style
+    final body = document.body;
+    if (body != null) {
+      // TODO: Modify our CSS to accomodate the styles output from dartdoc
+      body.querySelectorAll('.markdown').forEach((elem) {
+        if (!elem.classes.contains('markdown-body')) {
+          elem.classes.add('markdown-body');
+        }
+      });
+    }
+
+    // Extract breadcrumbs, because we need to tweak these on pub.dev
+    final breadcrumbs = document
+            .querySelector('header#title')
+            ?.querySelector('ol.breadcrumbs')
+            ?.children
+            .map((crumb) {
+          var href = crumb.querySelector('a')?.attributes['href'];
+          if (href != null && !_validLink(href)) {
+            href = null;
+          }
+          if (href != null) {
+            final u = Uri.tryParse(href);
+            if (u == null || u.shouldIndicateUgc) {
+              href = null; // breadcrumbs should never link externally!
+            }
+          }
+          return Breadcrumb(
+            title: crumb.text,
+            href: href,
+          );
+        }).toList() ??
+        <Breadcrumb>[];
+
+    final L = document.querySelector('#dartdoc-sidebar-left');
```

Comment:
Let's call this `left` (and similarly `R` -> `right`, `C` -> `content`), and we can call `sanitize(left?.innerHtml)` at the `DartDocPage` construction.